streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -134,6 +135,10 @@ public void onPartitionsRevoked(Collection<TopicPartition> assignment) {
                 initialized.set(false);
                 commitAll();
                 lastCleanMs = Long.MAX_VALUE; // stop the cleaning cycle until partitions are assigned
+                closeAllTasks();
```

Comment:
Ditto below, after closing the tasks we need to flush the state stores again before flushing the producer.